### ISSUE TITLE
Using GCP starter with Spring Data Rest results in Autoconfiguration error

### ISSUE THREAD

**Issue by zeagord** [spring-cloud/spring-cloud-gcp#673]
I was trying to Spring Cloud GCP for non storage services and my project uses a different data store. When I add spring-data-rest to my dependency. I am getting the following error.

```Error creating bean with name 'spannerKeyIdConverter' defined in class path resource [org/springframework/cloud/gcp/autoconfigure/spanner/SpannerKeyRestSupportAutoConfiguration.class]```

I think seeing the spring-data-rest, Spanner tries to add Auto-configuration. I tried disabling spanner but with no luck. 

```
spring.cloud.gcp.spanner.enabled=false
```
I had to exclude the autoconfigure classes to make it work. 

```
autoconfigure:
    exclude: org.springframework.cloud.gcp.autoconfigure.spanner.SpannerRepositoriesAutoConfiguration, org.springframework.cloud.gcp.autoconfigure.spanner.SpannerKeyRestSupportAutoConfiguration
---
```

Demo project could be found here: 
https://github.com/zeagord/springboot-gcp.git

**Comment by joaoandremartins** [comment]
@ChengyuanZhao mind taking a look? Thanks!

**Comment by ChengyuanZhao** [comment]
Taking a look!

**Comment by ChengyuanZhao** [comment]
This seems to work with the 1.0.0.BUILD-SNAPSHOT, but not the M3 being used. The difference seems to be in https://github.com/spring-cloud/spring-cloud-gcp/blob/eeed548ec3a8a82bfd158a6f40d52a94b26ea61e/spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/SpannerKeyRestSupportAutoConfiguration.java

`@ConditionalOnClass({ BackendIdConverter.class })` vs `@ConditionalOnClass({ BackendIdConverter.class, SpannerMappingContext.class})` 

I Balint might have come across an error like this before and fixed it since the M3 release.

**Comment by zeagord** [comment]
Appreciate it. I will keep it open for tracking it and close it with the next release.

**Comment by artembilan** [comment]
@zeagord ,

Thank you for using this project! Any your feedback is valuable

However...

>  I will keep it open for tracking it and close it with the next release.


The things don't work here this way. If we confirm issue and have a fix for it, we closed it with the link to the fix. You, as a reporter, don't have already a control over issue lifecycle. 

Saying that here we go!

Fixed via: https://github.com/spring-cloud/spring-cloud-gcp/commit/1a73d163737b1c1998c579b9e1f84c933012929e

Thank you for understanding.
